### ISSUE TITLE
Migrate to `org.kde.StatusNotifierItem` from `libayatana-appindicator`/`libappindicator-gtk3`

### ISSUE THREAD

**Issue by C0rn3j** [Taiko2k/Tauon#1316]
https://github.com/AyatanaIndicators/libayatana-appindicator/issues/22

Tauon cannot use GTK4 before this is done as the system tray feature relies on GTK3 to be available, and mixing 3.0 and 4.0 is not possible.

Example project that has done this already is Safe Eyes - https://github.com/slgobinath/SafeEyes/pull/558

**Comment by stkw0** [comment]
How it would work for people not using KDE (nor GNOME)?

**Comment by C0rn3j** [comment]
The DE needs to implement the spec - https://www.freedesktop.org/wiki/Specifications/StatusNotifierItem/ - even Xfce seems to do so based on a quick check, so I imagine everyone supports it one way or another.

**Comment by stkw0** [comment]
I was thinking specifically with people who uses a window manager instead of a DE.

**Comment by C0rn3j** [comment]
Window managers tend to have tray support?

Either way, same deal, instead of AppIndicator spec you need the StatusNotifierItem spec implemented.

**Comment by stkw0** [comment]
Yes. At least dwm did, and most wm I know or used had it (though I moved to KDE years ago). In any case, I guess it could be made in such a way that notifications are optional and Tauon can run without notifications support if the WM don't implement that spec.

**Comment by C0rn3j** [comment]
Note that this is not about notifications but the tray icon and its associated functions.

Tray icon is already optional and off by default.

Both the current indicator and the kde implementation over dbus(?) actually are both SNI (StatusNotifierItem) implementations... as you can see I have not looked too deep into this

**Comment by C0rn3j** [comment]
Dependent on #1332, but https://www.phoronix.com/news/SDL-System-Tray-Support seems to perhaps be the best option here. 

We'd ditch all the current tray implementations for an SDL native one, since Tauon is heavily based on SDL anyway.

**Comment by C0rn3j** [comment]
`(tauonmb:608088): libayatana-appindicator-WARNING **: 02:07:58.823: libayatana-appindicator is deprecated. Please use libayatana-appindicator-glib in newly written code.`

New warning too.

**Comment by C0rn3j** [comment]
We have now migrated to SDL native functions for the tray, so this will be automatically fixed when SDL 3.6.0 releases.

Keeping this open to remember to bump GTK 3 to GTK 4 (or tear it out completely) when that happens.
